air/src/execution_step/air/fail.rs

```diff
@@ -45,25 +49,32 @@ impl<'i> super::ExecutableInstruction<'i> for Fail<'i> {
     }
 }
 
+fn fail_with_scalar<'i>(scalar: &ast::ScalarWithLambda<'i>, exec_ctx: &mut ExecutionCtx<'i>) -> ExecutionResult<()> {
+    let (value, mut tetraplet) = resolve_ast_scalar_wl(scalar, exec_ctx)?;
+    // tetraplets always have one elements here and it'll be refactored after boxed value
+    let rtetraplet = tetraplet.remove(0);
```

Comment:
Maybe just shadow old `tetraplet`? Or make a meaningful name, if shadowing is not possible.